src/lightning/data/streaming/resolver.py

```diff
@@ -0,0 +1,340 @@
+        pass
+
+    _LIGHTNING_SDK_AVAILABLE = False
+
+
+@dataclass
+class Dir:
+    """Holds a directory path and possibly its associated remote URL."""
+
+    path: Optional[str] = None
+    url: Optional[str] = None
+
+
+def _resolve_dir(dir_path: Optional[Union[str, Dir]]) -> Dir:
+    if isinstance(dir_path, Dir):
+        return Dir(path=str(dir_path.path) if dir_path.path else None, url=str(dir_path.url) if dir_path.url else None)
+
+    if dir_path is None:
+        return Dir()
+
+    if not isinstance(dir_path, str):
+        raise ValueError(f"`dir_path` must be a `Dir` or a string, got: {dir_path}")
+
+    assert isinstance(dir_path, str)
+
+    if dir_path.startswith("s3://"):
+        return Dir(path=None, url=dir_path)
+
+    dir_path = _resolve_time_template(dir_path)
+
+    dir_path_absolute = str(Path(dir_path).absolute().resolve())
+
+    if dir_path_absolute.startswith("/teamspace/studios/this_studio"):
+        return Dir(path=dir_path_absolute, url=None)
+
+    if dir_path_absolute.startswith("/.project"):
+        dir_path_absolute = dir_path
+
+    if dir_path_absolute.startswith("/teamspace/studios") and len(dir_path_absolute.split("/")) > 3:
+        return _resolve_studio(dir_path_absolute, dir_path_absolute.split("/")[3], None)
+
+    if dir_path_absolute.startswith("/teamspace/s3_connections") and len(dir_path_absolute.split("/")) > 3:
+        return _resolve_s3_connections(dir_path_absolute)
+
+    if dir_path_absolute.startswith("/teamspace/datasets") and len(dir_path_absolute.split("/")) > 3:
+        return _resolve_datasets(dir_path_absolute)
+
+    return Dir(path=dir_path_absolute, url=None)
+
+
+def _match_studio(target_id: Optional[str], target_name: Optional[str], cloudspace: V1CloudSpace) -> bool:
+    if cloudspace.name is not None and target_name is not None and cloudspace.name.lower() == target_name.lower():
+        return True
+
+    if target_id is not None and cloudspace.id == target_id:
+        return True
+
+    if (
+        cloudspace.display_name is not None
+        and target_name is not None
+        and cloudspace.display_name.lower() == target_name.lower()
+    ):
+        return True
+
+    return False
+
+
+def _resolve_studio(dir_path: str, target_name: Optional[str], target_id: Optional[str]) -> Dir:
+    client = LightningClient(retry=False)
+
+    # Get the ids from env variables
+    cluster_id = os.getenv("LIGHTNING_CLUSTER_ID", None)
+    project_id = os.getenv("LIGHTNING_CLOUD_PROJECT_ID", None)
+
+    if cluster_id is None:
+        raise RuntimeError("The `cluster_id` couldn't be found from the environement variables.")
+
+    if project_id is None:
+        raise RuntimeError("The `project_id` couldn't be found from the environement variables.")
+
+    clusters = client.cluster_service_list_project_clusters(project_id).clusters
+
+    cloudspaces = client.cloud_space_service_list_cloud_spaces(project_id=project_id, cluster_id=cluster_id).cloudspaces
```

Comment:
Would you mind adding this to the project API of the SDK? we shouldn't need to expose lightning_cloud API calls here

Reply:
Yes, in future work.